src/Core/Core.idr

```diff
@@ -448,7 +448,7 @@ coreFail e = MkCore (pure (Left e))
 
 export
 wrapError : (Error -> Error) -> Core a -> Core a
-wrapError fe (MkCore prog) = MkCore $ prog <&> mapFst fe
+wrapError fe (MkCore prog) = MkCore $ prog >>= pure . mapFst fe
```

Comment:
This is fancier though, `<&>` for `>>= pure .`.

Reply:
Fancier in the sense it's not shipped by the previous version.

Reply:
Oh right

Reply:
If `<&>` is unavailable, then use `<$>`, it's definitely available. I.e.
```idris
wrapError fe (MkCore prog) = MkCore $ mapFst fe <$> prog
```
IMO, still better than `>>= pure .`